### DISCUSSION TITLE
Best way to store DB Connection Parameters for SQL Q&A

### DISCUSSION THREAD

**Discussion by ItsmeKernel** [FlowiseAI/Flowise#2784] [Q&A] [answered]
Hi! We are just getting started with FlowiseAI, and we are working on a chatbot to talk to our SQL Database. We'd like to ask the community on the best practices for storing/accessing the parameters to connect to our MySQL Server.

We've done some research, and we've only seen examples where this information is hardcoded in a JS function, as seen in the [official documentation]( https://docs.flowiseai.com/use-cases/sql-qna). Note that this article mentions that more information on this topic can be found in a now broken link.

We have a parent application that owns this database. It has a YAML configuration file with the parameters for connecting to the MySQL Server. Should we try to read the information from a JS Custom Function? Should we place the parameters in an `.env` file?

Thanks!

**Comment by HenryHengZJ** [comment]
One way to store parameters is to use the Global variables https://docs.flowiseai.com/using-flowise/variables

You can then get the variables using `$vars.variable` in the custom JS function

**Reply by ItsmeKernel** [reply]
Hi! Thanks for your reply!

We are looking for a solution that, ideally, doesn't involve hardcoding these values in FlowiseAI. Security is a concern, as our systems get pen-tested regularly.

**Reply by HenryHengZJ** [reply, marked as answer]
You can put those variables in a .env file, and using vars to pick up those values from .env file

**Reply by ItsmeKernel** [reply]
Thanks! I was thinking along those lines.
